server/consumer.go

```diff
@@ -399,6 +467,10 @@ type consumer struct {
 
 	// for stream signaling when multiple filters are set.
 	sigSubs []*subscription
+
+	// Priority groups
+	currentNuid string
```

Comment:
Can you please add comments for what this is for? Is this what identifies a unique pinned client? Does the absence of a value here mean unpinned?

Reply:
This is still vague and the ADR is light on details. What does the nuid *do*? Where does it come from? When does it change?

Reply:
For the pinning behaviour the server calculates a random id and gives that to the client who it decides is the pinned client.  That client will send the ID back in pull requests so we know which of the clients are pinned one later and can detect if old pulls come in etc.

Reply:
Thanks, can we document that in the comments? 😄

Reply:
ADR covers it here https://github.com/nats-io/nats-architecture-and-design/blob/main/adr/ADR-42.md#pinned_client-policy - discussed when to store the pin nuid etc

Reply:
@neilalexander I've added the links to ADR, renamed the field, and gave some short explanation in code. Let me know if that is enough.